lib/galaxy/tools/toolbox/panel.py

```diff
@@ -79,7 +79,10 @@ def to_dict(self, trans, link_details=False):
             link_details=link_details
         )
         for elt in self.elems.values():
-            section_elts.append(elt.to_dict(**kwargs))
+            if elt.__class__.__name__ == 'Tool' and toolbox:
```

Comment:
This was probably bad advice because of inheritance like you said,  I like ``hasattr(elt, "tool_type")`` better here - it is more Pythonic and it will mean the collection tools and such are still cached.

Reply:
that is a better indicator, thanks!